### PR TITLE
GLO_QC (ITS-TPC matching): extend Chi2 range, pT histograms to TH1D and adding Phi and Eta vs pT

### PR DESCRIPTION
**Pull request by jloemker**
@chiarazampolli , @njacazio this should not cause any troubles; I could execute the workflow and produce the ITSTPCmatchQC.root

### CODE REVIEW

**Comment by njacazio** [comment]
Hi @jloemker if this includes https://github.com/AliceO2Group/AliceO2/pull/10408 i can close that one and we merge everything with this one

**Comment by jloemker** [comment]
@njacazio , I just included your modifications to the void Efficiency(). And, as you can see, get a formatting error which I cannot solve because I don't understand it.

**Comment by njacazio** [comment]
> @njacazio , I just included your modifications to the void Efficiency(). And, as you can see, get a formatting error which I cannot solve because I don't understand it.

@jloemker great thanks ! I saw the log, seems to be like it is failing, but might be spurious?

**Comment by njacazio** [comment]
@chiarazampolli I think we are ready to merge, we will then open the pr on quality control

**Review by chiarazampolli — DISMISSED** [review]

**Comment by chiarazampolli** [comment]
The PR is fine with me, but please, check the "Formatting / PR formatting / line endings (pull_request_target) " test. 
I also cannot merge, so my approval is not enough...
Chiara

**Comment by njacazio** [comment]
@shahor02 can we merge this? I think it also needs approval before the test starts

**Comment by jloemker** [comment]
@chiarazampolli @shahor02, can we try to merge this ?
We need the extended Chi2 range and even more urgently the changes for the Pt differential results from TH1F to TH1D to get a proper result for the efficiency in Pt.
Thanks a lot !

**Comment by shahor02** [comment]
Will merge once the CIs are passed.

**Comment by njacazio** [comment]
Could it be that Johanna is a first time contributor and the CI don't start before approval? This PR is open since some time now and no CI is completed

**Review by shahor02 — DISMISSED** [review]
Apparently, this is the 1st PR, approuving conditionally to start the CI

**Comment by shahor02** [comment]
Hi @jloemker 
The error looks genuine:
```
6981:qc-task-GLO-MTCITSTPC]: Info in <TROOT::TEfficiency::CheckEntries>: Histograms are not consistent: passed bin content > total bin content
[6981:qc-task-GLO-MTCITSTPC]: Error in <TROOT::TEfficiency::CheckConsistency>: passed TEfficiency objects do not have consistent bin contents
[6981:qc-task-GLO-MTCITSTPC]: [07:24:30][FATAL] Something went wrong when defining the efficiency mFractionITSTPCmatchPhi from mPhiVsPtTPC and mPhiVsPt
```

**Comment by jloemker** [comment]
Hello @shahor02, 
I changed the binning from the TH2F for the efficiency Phi,Eta vs Pt. It builds on my machine and the first checks here went smooth, but I think you have to approve again in order to get all the checks. Sorry for the inconvenience !

**Review by shahor02 — DISMISSED** [review]

**Comment by chiarazampolli** [comment]
Hello @jloemker ,

Maybe i overlooked, but just changing the number of bins might simply mask the issue; the error indicates that the numerator is not a subsample of the denominator. Can you check how you fill?

Chiara

**Comment by chiarazampolli** [comment]
Ps: it is not a compilation error, but it happens at execution

**Comment by njacazio** [comment]
It has to do with the check in https://github.com/AliceO2Group/AliceO2/pull/10749/files#diff-592dbcc1211281791a869815906f3888bd979ce85f862b13be9fbc7e8ba5bef8R190

File: Detectors/GlobalTracking/src/MatchITSTPCQC.cxx
Comment (njacazio) [inline Detectors/GlobalTracking/src/MatchITSTPCQC.cxx]: Here the name of `mFractionITSTPCmatchPhiVsPt` is not correct (i.e. `mFractionITSTPCmatchPhi`) and it clashes with the other TEfficiency, could this be the reason?

Comment (jloemker) [reply Detectors/GlobalTracking/src/MatchITSTPCQC.cxx]: Thank you very much for pointing it out !
I really hope this fix works now - else I would exclude the  'mFractionITSTPCmatchPhiVsPt' for now, because we really need the TH1F to TH1D fix and the extended Chi2 range.

**Review by shahor02 — DISMISSED** [review]

**Comment by shahor02** [comment]
@jloemker still the same fatal in the QC

**Comment by jloemker** [comment]
Hello @shahor02,
i commented out the setEfficiency for Eta and Phi vs Pt which caused the error on runtime to finally get proper results for the asynch QC meetings.
The x additional changes in the other files are purely from formatting with Clang.
And again, my apologies for the several failed attempts !

**Review by shahor02 — APPROVED** [review]

**Comment by chiarazampolli** [comment]
Hello Johanna, 
I know that this PR is now closed, but I wanted to comment, trying to help. 
The failures in this PR mean that you fill the numerator and the denominator wrongly, ending up sometimes with more entries in a bin in the numerator than you have in the denominator. 
I would suggest to run the same command as in the CI (maybe @TimoWilken can help in this respect) locally, and debug.
Chiara

**Comment by TimoWilken** [comment]
Hi, if you want to run the CI tests locally, most of them just have O2 and O2Physics checked out locally (containing your code that you want to test), then do an `aliBuild build O2Suite -e ALIBUILD_O2_TESTS=true -e ALIBUILD_O2PHYSICS_TESTS=true`.

The "full system tests"/"fullCI" check is a bit more complicated; to execute that locally, you'd have to have Docker running (as we need a special container with e.g. the ROCm SDK to compile against GPU targets), then run `aliBuild build O2-full-system-test -e ALIBUILD_O2_TESTS=true -e ALIBUILD_O2PHYSICS_TESTS=true -a slc8_x86-64 --docker-image alisw/slc8-gpu-builder:latest --docker-extra-args=--tmpfs=/dev/shm:rw,size=8g,mode=1777`.

Let me know if you need any help with that.

**Pull request closed without merging**
